untrusted/urts/src/lib.rs

```diff
@@ -193,4 +269,53 @@ mod tests {
         let builder = EnclaveBuilder::new(b"").debug(true);
         assert_eq!(builder.debug as c_int, 1);
     }
+
+    #[test]
+    fn no_report_function_results_in_error() {
+        let enclave = EnclaveBuilder::new(ENCLAVE).create().unwrap();
+        let target_info = MaybeUninit::zeroed();
+        let target_info = unsafe { target_info.assume_init() };
+        assert_eq!(
+            enclave.create_report(Some(target_info)).unwrap_err(),
+            Error::NoReportFunction
+        );
+    }
+
+    #[test]
+    fn report_function_provides_report() {
+        let enclave = EnclaveBuilder::new(ENCLAVE)
+            .report_fn(Some(|enclave, target_info| {
+                let report = MaybeUninit::zeroed();
+                let mut report = unsafe { report.assume_init() };
+                let mut retval: sgx_status_t = sgx_status_t::SGX_SUCCESS;
+                let info = match target_info {
+                    Some(info) => info,
+                    None => ptr::null(),
+                };
+                let result =
+                    unsafe { ecall_create_report(**enclave, &mut retval, info, &mut report) };
+                match result {
+                    sgx_status_t::SGX_SUCCESS => match retval {
+                        sgx_status_t::SGX_SUCCESS => Ok(report),
+                        x => Err(Error::SgxStatus(x)),
+                    },
+                    x => Err(Error::SgxStatus(x)),
+                }
+            }))
```

Comment:
This looks like an impl we'd reuse, so how about we promote it to a `pub fn`?

Reply:
Re-worked this in https://github.com/mobilecoinfoundation/sgx/pull/11/commits/3af723a26b569caa64e7873071c37c4cc7aee359